diff --git a/apps/openmw/CMakeLists.txt b/apps/openmw/CMakeLists.txt
index 1c11825ef..a66c4d39b 100644
--- a/apps/openmw/CMakeLists.txt
+++ b/apps/openmw/CMakeLists.txt
@@ -148,7 +148,7 @@ if(OSG_STATIC)
     add_library(openmw_osg_plugins INTERFACE)
     foreach(_plugin ${USED_OSG_PLUGINS})
         string(TOUPPER ${_plugin} _plugin_uc)
-        list(APPEND _osg_plugins_static_files $<TARGET_FILE:${${_plugin_uc}_LIBRARY}>)
+        list(APPEND _osg_plugins_static_files ${${_plugin_uc}_LIBRARY})
         target_link_libraries(openmw_osg_plugins INTERFACE ${${_plugin_uc}_LIBRARY})
     endforeach()
     # We use --whole-archive because OSG plugins use registration.
@@ -158,7 +158,9 @@ if(OSG_STATIC)
 endif(OSG_STATIC)
 
 if (ANDROID)
-    target_link_libraries(openmw EGL android log z)
+    find_package(OSGPlugins REQUIRED COMPONENTS ${USED_OSG_PLUGINS})
+    list(APPEND OPENSCENEGRAPH_LIBRARIES ${OSGPlugins_LIBRARIES})
+    target_link_libraries(openmw EGL android log z ${OPENSCENEGRAPH_LIBRARIES} freetype jpeg png)
 endif (ANDROID)
 
 if (USE_SYSTEM_TINYXML)
